### PR TITLE
add docs for nested dynamic disks

### LINKED ISSUE #970: Add docs for nested dynamic disks
See the 23.3 release webinar for details and a nice example using the UK property prices dataset.

### PR DESCRIPTION
**Pull request by DanRoscigno**
Closes https://github.com/ClickHouse/clickhouse-docs/issues/970

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 30dd645b4f2bcffd1fb7d8a40a5040fa71d4a926 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49364/30dd645b4f2bcffd1fb7d8a40a5040fa71d4a926/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 1]
```diff
@@ -876,6 +882,48 @@ You could change storage policy after table creation with [ALTER TABLE ... MODIF
 
 The number of threads performing background moves of data parts can be changed by [background_move_pool_size](/docs/en/operations/server-configuration-parameters/settings.md/#background_move_pool_size) setting.
 
+### Nested Dynamic Storage
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:885]: btw is there a doc for non-nested dynamic disks? seems no, I did not find, then may add it here as well?

**Comment by DanRoscigno** [comment]
@kssenii I used type=url, I see from the tests that type=s3 is also supported.  Are there others that work with dynamic?

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 2]
```diff
@@ -882,11 +882,46 @@ You could change storage policy after table creation with [ALTER TABLE ... MODIF
...
+
 ### Nested Dynamic Storage
 
-This example query shows how to use a local disk for to cache data from a table stored
-at a URL.  Neither the cache disk nor the web storage is configured in the ClickHouse
-configuration files; both are configured in the ATTACH query settings.
+This example query builds on the above dynamic disk configuration and shows how to
+use a local disk to cache data from a table stored at a URL. Neither the cache disk
+nor the web storage is configured in the ClickHouse configuration files; both are
+configured in the ATTACH query settings.
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:924]:
```suggestion
configured in the CREATE/ATTACH query settings.
```

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 3]
```diff
@@ -882,11 +882,46 @@ You could change storage policy after table creation with [ALTER TABLE ... MODIF
 
 The number of threads performing background moves of data parts can be changed by [background_move_pool_size](/docs/en/operations/server-configuration-parameters/settings.md/#background_move_pool_size) setting.
 
+### Dynamic Storage
+
+This example query shows how to attach a table stored at a URL and configure the
+remote storage within the query. The web storage is not configured in the ClickHouse
+configuration files; all the settings are in the ATTACH query.
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:889]:
```suggestion
configuration files; all the settings are in the CREATE/ATTACH query.
```

**Comment by kssenii** [comment]
> @kssenii I used type=url, I see from the tests that type=s3 is also supported. Are there others that work with dynamic?

yes, any disk type can be configured as dynamic, even Local disk, but non-remote disks require `path` argument to be inside server config parameter `custom_local_disks_base_directory`, which has no default, e.g. will throw if local disk is created as dynamic disk and there is no such value in config.
And as an example of nested disk you can also use "encrypted" disk instead of "cache" disk.

**Review by kssenii — APPROVED** [review]

**Pull request merged**